File: transfer/santri.js
const mysql = require ('mysql2');


// Konfigurasi koneksi ke database sumber (Santri)
    const sourceDB = mysql.createConnection({
        host: 'localhost',
        user: 'root',
        password: 'Pastibisa',
        database: 'Santri',
    });

    // Konfigurasi koneksi ke database tujuan (pzn)
    const targetDB = mysql.createConnection({
        host: 'localhost',
        user: 'root',
        password: 'Pastibisa',
        database: 'goSantri',
    });

    // Query data dari database sumber
    sourceDB.query('SELECT * FROM Santri', (err, results) => {
        if (err) {
            console.error('Error querying source database:', err);
            return;
        }

        // Memasukkan data ke database tujuan
        const dataToInsert = results.map((row) => [row.nama, row.user_id, row.hp, row.email, row.gender, row.alamat, row.angkatan, row.jurusan, row.minat, row.status]);
        targetDB.query('INSERT INTO Santri (nama, user_id, hp, email, gender, alamat, angkatan, jurusan, minat, status) VALUES ?', [dataToInsert], (err) => {
            if (err) {
                console.error('Error inserting data into target database:', err);
            } else {
                console.log('Data successfully transferred to target database.');
            }
            // Tutup koneksi ke kedua database
            sourceDB.end();
            targetDB.end();
        });
    });

File: model/entity/Santri.go
package entity

type Santri struct {
	ID          int64         `json:"id"`
	Nama        string        `json:"nama"`
	HP          int64         `json:"hp"`
	Email       string        `json:"email"`
	Gender      string        `json:"gender"`
	Alamat      string        `json:"alamat"`
	Angkatan    int           `json:"angkatan"`
	Jurusan     string        `json:"jurusan"`
	Minat       string        `json:"minat"`
	Status      string        `json:"status"`
	Attachments []Attachments `gorm:"foreignKey:user_id" json:"attachments"`
}

func (Santri) TableName() string {
	return "Santri"
}

// Create Santri

type ForCreateSantri struct {
	ID       int64  `json:"id"`
	UserID   int64  `json:"user_id"`
	Nama     string `json:"nama" binding:"required"`
	HP       int64  `json:"hp"`
	Email    string `json:"email"`
	Gender   string `json:"gender" binding:"required"`
	Alamat   string `json:"alamat" binding:"required"`
	Angkatan int    `json:"angkatan"`
	Jurusan  int    `json:"jurusan"`
	Minat    int    `json:"minat"`
	Status   int    `json:"status"`
}

type CreateSantri struct {
	Nama     string `json:"nama"`
	UserID   int64  `json:"user_id"`
	HP       int64  `json:"hp"`
	Email    string `json:"email"`
	Gender   string `json:"gender"`
	Alamat   string `json:"alamat"`
	Angkatan int    `json:"angkatan"`
	Jurusan  *int   `json:"jurusan"`
	Minat    *int   `json:"minat"`
	Status   *int   `json:"status"`
}

func (CreateSantri) TableName() string {
	return "Santri"
}

// UPDATE

//type Gender string
//
//const (
//	Pria   Gender = "Pria"
//	Wanita Gender = "Wanita"
//)
